tests/test_config.py

```diff
@@ -152,6 +152,19 @@ def test_from_pretrained_cache_dir_remote(self):
             PeftConfig.from_pretrained("ybelkada/test-st-lora", cache_dir=tmp_dirname)
             assert "models--ybelkada--test-st-lora" in os.listdir(tmp_dirname)
 
+    @parameterized.expand(ALL_CONFIG_CLASSES)
+    def test_save_pretrained_with_runtime_config(self, config_class):
+        r"""
+        Test if the config correctly removes runtime config when saving
+        """
+        with tempfile.TemporaryDirectory() as tmp_dirname:
+            for model_name, revision in PEFT_MODELS_TO_TEST:
+                cfg = config_class.from_pretrained(model_name, revision=revision)
```

Comment:
This test does not do what it appears to be doing. No matter the `config_class`, ` config_class.from_pretrained` will always return a `LoraConfig` instance because the stored config comes from `LoraConfig`. This is a bit unintuitive but hard to change now without breaking existing code.

Reply:
Ahh, yeah, that's what I suspected. Thanks for verifying. I'll add a note about that for future explorers.